stdlib/websub/src/main/ballerina/websub/hub_caller_actions.bal

```diff
@@ -272,53 +257,51 @@ function buildSubscriptionChangeRequest(@sensitive string mode,
+            }
+            error webSubError = error(WEBSUB_ERROR_CODE, { message : errorMessage });
             return webSubError;
-        }
-        http:Response httpResponse => {
-            int responseStatusCode = httpResponse.statusCode;
-            if (responseStatusCode == http:TEMPORARY_REDIRECT_307
-                    || responseStatusCode == http:PERMANENT_REDIRECT_308) {
-                if (remainingRedirects > 0) {
-                    string redirected_hub = httpResponse.getHeader("Location");
-                    return invokeClientConnectorOnRedirection(redirected_hub, mode, subscriptionChangeRequest,
-                                                                httpClientEndpoint.config.auth, remainingRedirects - 1);
-                }
-                map errorDetail = { message : "Redirection response received for subscription change request"
-                                        + " made with followRedirects disabled or after maxCount exceeded: Hub ["
-                                        + hub + "], Topic [" + subscriptionChangeRequest.topic + "]" };
-                error subscriptionError = error(WEBSUB_ERROR_CODE, errorDetail);
-                return subscriptionError;
-            } else if (!isSuccessStatusCode(responseStatusCode)) {
-                var responsePayload = httpResponse.getTextPayload();
-                string errorMessage = "Error in request: Mode[" + mode + "] at Hub[" + hub + "]";
-                match (responsePayload) {
-                    string responseErrorPayload => { errorMessage = errorMessage + " - " + responseErrorPayload; }
-                    error payloadError => {
-                        string errCause = <string> payloadError.detail().message;
-                        errorMessage = errorMessage + " - Error occurred identifying cause: " + errCause;
-                    }
-                }
-                error webSubError = error(WEBSUB_ERROR_CODE, { message : errorMessage });
-                return webSubError;
-            } else {
-                if (responseStatusCode != http:ACCEPTED_202) {
-                    log:printDebug("Subscription request considered successful for non 202 status code: "
-                                    + responseStatusCode);
-                }
-                SubscriptionChangeResponse subscriptionChangeResponse = {hub:hub, topic:topic, response:httpResponse};
-                return subscriptionChangeResponse;
+        } else {
+            if (responseStatusCode != http:ACCEPTED_202) {
+                log:printDebug("Subscription request considered successful for non 202 status code: "
+                                + responseStatusCode);
             }
+            SubscriptionChangeResponse subscriptionChangeResponse = {hub:hub, topic:topic, response:response};
+            return subscriptionChangeResponse;
         }
+    } else {
```

Comment:
Yes, Still type guard is not supporting to have only the else clause without type test expression. Since we have used else if clause, the function complains about not having default return at compile time. In order to overcome that limitation, else clause was added. This should be removed in the iteration where else clause is supported in the type guard.

Reply:
Was wondering if we should just return without an else block, but this looks cleaner, so +1.